src/T2DMap.cpp

```diff
@@ -724,6 +783,46 @@ void T2DMap::paintEvent(QPaintEvent* e)
 
     mRX = qRound(mRoomWidth * ((xspan / 2.0) - ox));
     mRY = qRound(mRoomHeight * ((yspan / 2.0) - oy));
+    QFont roomVNumFont = mpMap->mMapSymbolFont;
+    bool isFontBigEnoughToShowRoomVnum = false;
+    if (mShowRoomID) {
+        /*
+         * If we are to show the room Id numbers - find out the number of digits
+         * that we will need to use; actually, knowing the digit count is also
+         * useful for the room selection widget so perform this check EVERY time.
+         * TODO: Eventually move this check to the TArea class and just redo it
+         * when areas' room content changes.
+         */
+        int maxUsedRoomId = 0;
+        QSetIterator<int> itRoomId(playerArea->getAreaRooms());
+        while (itRoomId.hasNext()) {
+            maxUsedRoomId = qMax(maxUsedRoomId, itRoomId.next());
+        }
+        mMaxRoomIdDigits = static_cast<quint8>(QString::number(maxUsedRoomId).length());
+
+        QRectF roomTestRect;
+        if (playerArea->gridMode) {
+            roomTestRect = QRectF(0, 0, static_cast<qreal>(mRoomWidth), static_cast<qreal>(mRoomHeight));
+        } else {
+            roomTestRect = QRectF(0, 0, static_cast<qreal>(mRoomWidth) * rSize, static_cast<qreal>(mRoomHeight) * rSize);
+        }
+        QFont roomVNumFont = mpMap->mMapSymbolFont;
+        static quint8 roomVnumMargin = 10;
+        roomVNumFont.setUnderline(true);
+        roomVNumFont.setOverline(true);
+        roomVNumFont.setBold(true);
+#if QT_VERSION >= 0x050a00
```

Comment:
```suggestion
#if (QT_VERSION >= QT_VERSION_CHECK(5, 10, 0))
```

Reply:
I guess it is safe to use that NOW but `QT_VERSION_CHECK` was not present in the early versions of Qt 5 IIRC...

Reply:
Done.